pkg/specgen/namespaces.go

```diff
@@ -215,12 +245,17 @@ func ParseUserNamespace(ns string) (Namespace, error) {
 func ParseNetworkNamespace(ns string) (Namespace, []string, error) {
 	toReturn := Namespace{}
 	var cniNetworks []string
+	// Net defaults to Slirp on rootless
+	if rootless.IsRootless() {
```

Comment:
This completely overrides parsing on rootless, so it'll break rootless `--net=host`